### PR TITLE
text_detection_demo: fix auto resize 

### PR DESCRIPTION
**Pull request by ivikhrev**
GPU plugin doesn't support auto resize feature. Make opencv resize as default.
p.s  `text-recognition-0015-encoder` model will fail with stack overflow error.

### CODE REVIEW

File: demos/text_detection_demo/cpp/src/cnn.cpp [hunk 1]
```diff
@@ -66,16 +66,19 @@ Cnn::Cnn(
 
     // we'd like to pass input image (NCWH, u8) to model input
```
Comment (vladimir-dudnik) [inline demos/text_detection_demo/cpp/src/cnn.cpp:67]:
```suggestion
    // we'd like to pass input image (NHWC, u8) to model input
```

File: demos/text_detection_demo/cpp/src/cnn.cpp [hunk 2]
```diff
@@ -64,18 +64,21 @@ Cnn::Cnn(
...
+    // we'd like to pass input image (NHWC, u8) to model input
     // and let OpenVINO do necessary conversions
+
     ppp.input().tensor()
         .set_element_type(ov::element::u8)
-        .set_layout({ "NHWC" })
-        .set_spatial_dynamic_shape();
+        .set_layout({ "NHWC" });
 
-    ppp.input().preprocess().convert_layout(input_layout);
```
Comment (vladimir-dudnik) [inline demos/text_detection_demo/cpp/src/cnn.cpp:74]: why removed? will this work with IR v10?

**Review by vladimir-dudnik — APPROVED** [review]

File: demos/text_detection_demo/cpp/src/cnn.cpp (hunk 2 above)
Comment (ivikhrev) [reply demos/text_detection_demo/cpp/src/cnn.cpp:74]: layout conversion step is added when you set layout to tensor:
```
   ppp.input().tensor()
        .set_element_type(ov::element::u8)
        .set_layout({ "NHWC" });
```
We can see it in log:
```
[ INFO ] Preprocessor configuration:
[ INFO ] Input "Placeholder":
    User's input tensor: {1,768,1280,3}, [N,H,W,C], u8
    Model's expected tensor: {1,3,768,1280}, [N,C,H,W], f32
    Implicit pre-processing steps (2):
      convert type (f32): ({1,768,1280,3}, [N,H,W,C], u8) -> ({1,768,1280,3}, [N,H,W,C], f32)
      convert layout [N,C,H,W]: ({1,768,1280,3}, [N,H,W,C], f32) -> ({1,3,768,1280}, [N,C,H,W], f32)
```
no need to set it in preprocess.

**Review by Wovchena — APPROVED** [review]
You shouldn't set a default value for `auto_resize` in `Cnn` and the children because you never use that default value

File: demos/text_detection_demo/cpp/src/cnn.cpp (hunk 2 above)
Comment (vladimir-dudnik) [reply demos/text_detection_demo/cpp/src/cnn.cpp:74]: that was to make that obvious, but ok

**Comment by vladimir-dudnik** [comment]
ignore not relevant issues

**Pull request merged**